### PR TITLE
Solutions to Django Common Errors in Web Development

### LINKED ISSUE #5922: Solutions to Common Django Errors in Web Development
**Suggested article title**

Solutions to Common Django Errors in Web Development

**Proposed topic introduction**

Many times, people get stuck on the way, not only on coding related development but also several other development. This may be as a result of failure to follow instructions, installation error, and various other major or minor reasons. In this article, we shall major on Django errors during web development.
It is almost impossible to reach your destination in web development using any other framework including django without getting stuck on the way. At times there must be an emergence of error. Considering web development, it can be as a result of wrong path, installation, procedures and just ignorance. In this article, I'll make it easy for youby clearly pointing to some common errors associated with django, explaining to you step by step how we can solve these errors to keep going and reach our destination. Happy reading.

**Key takeways**

- 404 page not found error
- No reverse match error
- No module named error
- Template does not exist error
- Djanho-admin not recognized as internal and external command
- Djangi not able to find static files
- NameError
- Internal server error

**Article quality**

In my article, I'll be solving different errors in one article making it easier for the reader to easily find the error and solve it. Not as other articles, in my article, I will be considering several causes of a particular error and give solutions to them since an error can be brought about by many inconveniences. The aim of my article is to make it easy for programmers to reach their destination.

**References**

N/A

### PR DESCRIPTION
**Pull request by OkothAlvan**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #5922 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by srishilesh** [comment]
@OkothAlvan 

Welcome to EngEd community!

Your PR has entered our reviewing queue. Please wait until a peer-reviewer pings you for revisions to be made.

Meanwhile, we suggest you to go through the published articles or approved PRs to see how the other authors have formatted their articles to avoid any potential closure of PRs.

Thanks.

**Comment by OkothAlvan** [comment]
Hi.
It's okay.

**Comment by lalith1403** [comment]
plagiarism check done ✔️

File: content/articles/solutions-to-django-common-errors-in-web-development/index.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+```html
+{% load static %}
+
+<html lang="en">
+    <head>
+    ...
+```
+#### Handling name error
+Any name you would want to use or refer to later should be imported in settings.py. For example, an error like 
+```‘DIRS’:[‘os’ is not defined, ```
```
Comment (Jethro-magaji) [inline content/articles/solutions-to-django-common-errors-in-web-development/index.md:129]: There's an error with the formatting here.

**Review by Jethro-magaji — CHANGES_REQUESTED** [review]
Hi! @OkothAlvan 
Well done for such an amazing article. However, there're a lot of corrections you need to make.
Please, run your article on tools like Grammarly and ProWritingAid for better editing.
Visit the EngEd new contributor's docs [here](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md) to guide you.
Here's a [link](https://github.com/section-engineering-education/engineering-education/pull/5924/commits/c081a0c67bd777e21d60815634c955e5565d9f00) to a PR with little or no revision to guide you on formatting. However, if you have any questions or need help, you can reach out to me on the EngEd slack channel.
Your author bio isn't properly formatted in the correct markdown format, please work on it.
Thanks.

File: content/articles/solutions-to-django-common-errors-in-web-development/index.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
+### Solutions to Common Django Errors in Web Development
+
+
+#### Introduction
```
[the remark this answers is not in the document]
Comment (OkothAlvan) [reply content/articles/solutions-to-django-common-errors-in-web-development/index.md]: Corrected.

File: content/articles/solutions-to-django-common-errors-in-web-development/index.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+On several occasions, people get stuck on the way while undertaking different tasks or procedures. Programming is not left out in this. We sometimes get stuck in coding as a result of failure to follow instructions, installation error, and many other major or minor reasons. In this article, we shall major on Django errors during web development.
+
+It is almost impossible to reach your destinations in web development using any framework with Django included without getting stuck at some point on the way. This problem specifically in web development can result from the wrong path, installation, procedures, and sometimes just ignorance. In this article, I will make your coding easy by clearly pointing to some common errors associated with Django, explaining step by step how to solve them to make sure we arrive at our destination. Happy reading!
+By the end of this tutorial, the reader should be able to solve the following:
```
[the remark this answers is not in the document]
Comment (OkothAlvan) [reply content/articles/solutions-to-django-common-errors-in-web-development/index.md]: Corrected.

File: content/articles/solutions-to-django-common-errors-in-web-development/index.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+
+By the end of this tutorial, the reader should be able to solve the following:
+  
+* 404 page not found error 
+* No module named error
+* Django-admin is not recognized as an internal or external command
+* Django is not able to find static files
+* Name error
+Internal server error
```
[the remark this answers is not in the document]
Comment (OkothAlvan) [reply content/articles/solutions-to-django-common-errors-in-web-development/index.md]: No. Let me eliminate it.

[the remark this answers is not in the document]
Comment (OkothAlvan) [reply content/articles/solutions-to-django-common-errors-in-web-development/index.md]: Corrected.

File: content/articles/solutions-to-django-common-errors-in-web-development/index.md [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+
+#### Prerequisites
+To have a better understanding of this article, the reader should:
+* Know how to create websites using the Django framework
+* Have a good working machine with the preferred IDE set up
```
[the remark this answers is not in the document]
Comment (OkothAlvan) [reply content/articles/solutions-to-django-common-errors-in-web-development/index.md]: Corrected.

File: content/articles/solutions-to-django-common-errors-in-web-development/index.md [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+3. Handling Django-admin is not recognized as an internal or external command
+
+4. Handling Django is not able to find static files
+
+5. Handling name error
+
+6. Further learning
+
+#### Handling 404 page not found error
+This means that the particular folder you are looking for, for example, home.html is not present in the folder. Admittedly, the template folder stores all HTML files in Django.
```
[the remark this answers is not in the document]
Comment (OkothAlvan) [reply content/articles/solutions-to-django-common-errors-in-web-development/index.md:37]: Corrected

**Comment by OkothAlvan** [comment]
Hi @Jethro-magaji.
I have gone through your preferred changes and made changes to all of them,
please confirm.

**Comment by Jethro-magaji** [comment]
Hi! @OkothAlvan 
Please go over the links I dropped in the comment prev.
Everything is not in place, please take your time to make proper corrections. 
Thanks.

**Comment by OkothAlvan** [comment]
Hello.
It's okay.

**Comment by OkothAlvan** [comment]
Hello  ***@***.****
*Having gone through earlier published articles, I have done some changes
to my article, please let me know if there could still be corrections
needed. If so, I would appreciate it if you point them out.*

**Comment by Jethro-magaji** [comment]
Hi! @OkothAlvan 
The code blocks aren't properly arranged within the backticks, as a result this renders other texts as code. 
Again your table of contents doesn't link to the headings, please refer to this PR [here](https://github.com/section-engineering-education/engineering-education/pull/6210/commits/acc44893d4ec258f4e077b9ce8e3e2ca2b9d6a9f) as a guide.
Thanks.

**Comment by OkothAlvan** [comment]
Thank you for the clarification.
Noted.

**Comment by OkothAlvan** [comment]
Hello @jethro-magaji.
I have done thr above mentioned corrections. Thanks.

**Comment by Jethro-magaji** [comment]
Hi! @OkothAlvan 
Well done for the edits, but your Author file isn't properly formatted. The author bio file should be renamed to `index.md`. Here's an [example](https://github.com/section-engineering-education/engineering-education/tree/master/content/authors/jairus-onkundi).

**Comment by OkothAlvan** [comment]
Noted.

**Comment by Jethro-magaji** [comment]
@OkothAlvan 
Check your author image, and see if it's properly placed in your name folder.

**Comment by OkothAlvan** [comment]
noted

**Comment by OkothAlvan** [comment]
Hello @jethro-magaji.
I have a hard task relocating the avatar.

**Comment by Jethro-magaji** [comment]
You don't need to relocate it, just delete the current one and add the image to your name directory.

**Comment by OkothAlvan** [comment]
I tried that but there wasn't an option for delete or edit. I can only
view. Let me make another trial.

**Comment by OkothAlvan** [comment]
Hi @jethro-magaji.
I hope everything is now right.

File: content/authors/alvan-wazecky/index.md [hunk 7]
```diff
@@ -0,0 +1,7 @@
+---
+title: Alvan Wazecky
+type: authors
+images:
+  - url: /engineering-education/authors/alvan-wazecky/avatar.jpg
```
Comment (Jethro-magaji) [inline content/authors/alvan-wazecky/index.md:5]: Change "avatar.jpg" to "avatar.png" because the new image path is png.

Comment (OkothAlvan) [reply content/authors/alvan-wazecky/index.md]: Done

**Comment by OkothAlvan** [comment]
I have corrected that.

**Comment by ahmadmardeni1** [comment]
This article is really light on substance.
Going to close it as of now to clear the queue. Take your time and work on it. @OkothAlvan 
cc @Jethro-magaji

**Comment by OkothAlvan** [comment]
why do you prefer to close after all the corrections done and been in final
stage.

**Comment by OkothAlvan** [comment]
This is worrying.

**Pull request closed without merging**